### DISCUSSION TITLE
Exclude files not working or im doing something wrong

### DISCUSSION THREAD

**Discussion by g1b3r** [scolladon/sfdx-git-delta#615] [Q&A]
Hi 

I want to exclude for example BoatDataService class in script i passing during package.xml generation 

EXCLUDING BoatDataService

{
  "error": "'BoatDataService' file does not exist",
  "output": ".",
  "success": false,
  "warnings": []
}
im pretty sure it is here. 

I tried to exclude using 

sfdx sgd:source:delta --to HEAD --from $(latestBuildCommit) -o . -i ${{ parameters.excludeMetadata }} (when im hardcoding this is not working too :( )

**Comment by g1b3r** [comment]
-i "./force-app/main/default/classes/BoatDataService.cls" this is generating a empty package.xml :(

**Reply by scolladon** [reply]
Hi @g1b3r 

the `-i` parameter takes a [file path](https://github.com/scolladon/sfdx-git-delta#exclude-some-metadata-only-from-destructivechangesxml) as a parameter. 
The file has to follow the gitignore syntax.

So you could write a `.sgdignore` file with this content (or so):
```

BoatDataService.cls
```

And use the file with the command:
```sh
$ sfdx sgd:source:delta --to HEAD --from <latest-build> -o . -i .sgdignore
```
